### ISSUE TITLE
Team 4 Feature Ticket sprint 2 - Attack effects of the main player character

### ISSUE THREAD

**Issue by LIU-Evelyn** [UQdeco2800/2021-ext-studio-2#136]
# Description
This feature is dedicated to the realization of the protagonist of the game, including the protagonist’s aesthetic image and the change of the image when doing actions. We need to consider the interactions it needs to perform when avoiding obstacles or picking up props, and how these interactions will affect its appearance when moving along the path. Therefore, it is also necessary to determine the different images of the main character due to changes in their action (state) during the interaction process. Each image should clearly correspond to a different interaction state.

# SubFeature/component

- Basic image - The image type of the main player character (person/car, etc.)
- The image when performing actions: Moving, Running, Jumping, and Crouching (if the image is a human); Accelerated and Break movement (if the image is a car, flying saucer, etc.)
- Image when picking up items/props - if he needs to pick up etc.
- Visualisation of characters image in different statuses (if needed to be changed) due to buffs, debuffs, or player status

# Examples
- The player should be able to avoid obstacles by moving laterally. (Or pick up the buff and use it)
- The aesthetic appearance of the protagonist can be selected among several different characters/models (late development stage).
- The player's possible image changes when getting buffs or interacting with obstacles.

# Dependencies
- Obstacles implementation affects the realization of the player's interaction with obstacles.  （Team 8）#14
- The image of the character depends on the interaction and the status of the character while running.  （Team 2 & Team 1) #9, #7
- Player buffs and debuffs (Team 3) #10

# Milestones
Implementing Main Character In-game interactions and animations:

- [ ] #111
- [ ] #112
 
# Extra add-ons (Future Sprint Goals)

1. Design and integrate player movements and animations for item pickups [Future]
2. Design and integrate animations for player buffs and debuffs [Future]
3. Dynamically changing player status and animations based on player health. [Future]
4. Dynamically changing player movement speed and attack strength based on player health. [Future]

# Members

- Prateek Kumar Gupta - @prateek45
- Rohith Palakirti - @kprohith 
- Jiacheng Ni - @RecklessPotcover (Alex)
- Jescy Xu - @enmmjc 
- Yingxin Liu - @LIU-Evelyn

**Comment by katrinafu1** [comment]
Hi team4, team 1 will not be implementing the weapon pickup in this sprint, as we have already got enough features and coders do not have enough time to implement this in sprint2. And btw way our props store is doing a shield feature, which might need you guys to switch to the shield protected mode after the player bought the shield in the prop store. Of coz our designers will draw the character with a shield image. Thanks! If you have any questions you can contact our coder, thanks! @PaddyZz  @Yonarp

**Comment by lisslulay** [comment]
Hi, this is team8.
We are doing a missile attack on the spaceship boss.
One of your designers asked us if we would set up obstacles based on the character's crouch pose. We made relevant attempts but failed. We found that this is because the character crouch only changed the character's animation, but did not change the actual height of the character. As shown in this video, no matter what pose the character does, his height remains the same.

https://user-images.githubusercontent.com/78423568/133817016-96c1968e-42ef-44d8-ab13-d46b2bd74305.mp4


Maybe you can try to change the height of the character when the crouch pose event is triggered. We will temporarily set the missile according to the height of the character. If you make the corresponding change, could you contact us?

**Comment by kprohith** [comment]
> Hi, this is team8.
> We are doing a missile attack on the spaceship boss.
> One of your designers asked us if we would set up obstacles based on the character's crouch pose. We made relevant attempts but failed. We found that this is because the character crouch only changed the character's animation, but did not change the actual height of the character. As shown in this video, no matter what pose the character does, his height remains the same.
> 
>  bandicam.2021-09-17.23-36-02-315.mp4 
> Maybe you can try to change the height of the character when the crouch pose event is triggered. We will temporarily set the missile according to the height of the character. If you make the corresponding change, could you contact us?

Yes, will look into it and let you know. In the last sprint, we only implemented the animations.
